doc: clarify `length` param in `buffer.write` (#32119)
nodejs/node: doc/api/buffer.md

* `string` {string} String to write to `buf`.
* `offset` {integer} Number of bytes to skip before starting to write `string`.
  **Default:** `0`.
* `length` {integer} Maximum number of bytes to write (written bytes will not
  exceed `buf.length - offset`). **Default:** `buf.length - offset`.

>   If the buffer has sufficient space from the offset, the string is written upto

Reviewer: ```suggestion
  If the buffer has sufficient space from the offset, the string is written up to
```

Author: @addaleax, thanks. Fixed it.

  up to `length`. If the buffer is short in space, only `buf.length - offset`

>   bytes are written.

Reviewer: @HarshithaKP The last three sentences all appear to be saying the same thing. Can I suggest that everything after the `**Default:**` can be removed, and instead by explained through examples below?

(Also, `**Default:**` should be the last item in the description here anyway.)

Author: @addaleax, thanks for the suggestion. Added example to it. PTAL.

const len = buf.write('\u00bd + \u00bc = \u00be', 0);

console.log(`${len} bytes: ${buf.toString('utf8', 0, len)}`);

> // Prints: 12 bytes: ½ + ¼ ¾

Reviewer: The output includes the `=` sign, though.

Author: Mistakenly edited file in two tabs. Got deleted.

* `string` {string} String to write to `buf`.
* `offset` {integer} Number of bytes to skip before starting to write `string`.
  **Default:** `0`.
* `length` {integer} Number of bytes to write (written bytes will not exceed

>   `buf.length - offset`). **Default:** `buf.length - offset`.

Reviewer 2: ```suggestion
* `length` {integer} Maximum number of bytes to write (written bytes
   will not exceed `buf.length - offset`). **Default:**
   `buf.length - offset`.
```

Author: @jasnell, added your suggestion. PTAL.

* `string` {string} String to write to `buf`.
* `offset` {integer} Number of bytes to skip before starting to write `string`.
  **Default:** `0`.

> * `length` {integer} Number of bytes to write.

Reviewer 3: ```suggestion
* `length` {integer} Number of bytes to write (written bytes will not exceed
  `buf.length - offset`). **Default:** `buf.length - offset`.
```

Author: @BridgeAR, thanks. Included your suggestion. PTAL.
